raft.go

```diff
@@ -369,12 +369,12 @@ func (rn *Node) Destroy() error {
 	}
 	rn.logger.Debug("Successfully removed self from canoe cluster")
 
-	if rn.running {
+	if rn.running.IsSet() {
```

Comment:
If two goroutines can be running this it's still racey since they might both observe it being set and run this section.

Even if you are sure this can only be run by a single goroutine, it's still potentially racey because other goroutines might still get a response to `node.isHealthy()` of true even after the stop chan is closed and transport is partially shutdown. That might not cause actual problems in practice but it probably violates assumptions about what is safe to assume and what isn't since none are documented in the methods.